docs/2_2_common_mechanisms.adoc

```diff
@@ -330,428 +338,335 @@ Set <<parameter,`parameters`>>, <<input,`inputs`>>, and <<start>> values, and re
-FMI Design meeting: remark that set interval functions are informative to design model based controllers.
-The environment is the one that will tick.
-]_
+Aperiodic clocks::
+are time-based clocks where the next interval is unchangeably known right after the clock just ticked.
+<<fmi3GetInterval>> must be called to retrieve the first interval in *Initialization Mode* and every next clock interval in every *Event Mode* if and only if the corresponding clock ticked.
+Calling <<fmi3GetShift>> is not allowed.
 
-A <<clock>> is <<strict,strictly>> <<periodic>> if latexmath:[\Delta T] is fixed for an FMU and cannot be changed.
-The <<clock>> is periodic, if latexmath:[\Delta T] can be defined before calling `fmi3EnterInitializationMode` via <<fmi3SetIntervalDecimal>> or <<fmi3SetIntervalFraction>>.
-The <<clock>> is <<periodic,aperiodic>> if latexmath:[\Delta T] is not constant during a simulation run (<<fmi3SetIntervalDecimal>> or <<fmi3SetIntervalFraction>> can be called for that <<clock>> during *Event Mode*).
+Countdown clocks::
+are time-based clocks where the next interval is not yet known right after the clock just ticked, forcing the importer to call <<fmi3GetInterval>> in *Initialization Mode* and in every *Event Mode*.
+The return argument <<qualifier>> of <<fmi3GetInterval>> is used to indicate if the next interval is already known.
+Calling <<fmi3GetShift>> is not allowed.
 
-===== Connecting Clocked FMUs [[connecting-clocked-fmus]]
+Triggered clocks::
+tick unpredictably.
 
-When connecting imported FMUs, <<input>> and <<output>> variables can readily be connected only when they are defined on a <<clock>> with identical properties.
-If this is not possible, an explicit cast of one <<clock>> to another <<clock>> is typically defined in the environment.
+Triggered input clocks::
+are activated with <<fmi3SetClock>> by the importer in *Event Mode*.
 
-_[TODO: What is a clock cast? If a clock cast can be done by the environment anyway, why mention which inputs/outputs can be connected?_
-_Remove the entire section: we do not want to say: don't be stupid.]_
+Triggered output clocks::
+are activated within the FMU and the importer must call <<fmi3GetClock>> in *Event Mode*.
+There are two mechanisms for an FMU to communicate to the importer that a triggered output clock is about to tick:
+ * <<early-return,early return>> from a <<fmi3DoStep>> with <<clocksAboutToTick, `clocksAboutToTick == fmi3True`>>, and
+ * calling <<fmi3CallbackIntermediateUpdate>> with <<clocksTicked, `clocksTicked == fmi3True`>>.
 
-The environment can evaluate required <<clock>> casts for connections using the information on the <<clockReference>> attribute of assigned variables _[i.e. the variables that are assigned to a <<clock>> based on <<modelDescription.xml>> information]_.
+Input clocks::
+[[inputClock, `input clock`]]All clocks with <<causality>> <<input>> are collectively called <<inputClock>>.
```

Comment:
Is this really true?  From the table above I understand that only _triggered_ clocks with causality input are called "input clocks"

Reply:
We will leave this as is, we did this on purpose to use input clocks in the text. If we end up not using it, we can remove this definition later.

Reply:
... and also: outputClocks may additionally depend on inputClocks (subclocks).